cmd/minishift/cmd/start.go

```diff
@@ -303,16 +309,21 @@ func initSubscriptionManagerFlags() {
 
 // clusterUp downloads and installs the oc binary in order to run 'cluster up'
 func clusterUp(config *cluster.MachineConfig) {
-	oc := cache.Oc{
+	oc := &cache.Oc{
 		OpenShiftVersion:  config.OpenShiftVersion,
 		MinishiftCacheDir: filepath.Join(constants.Minipath, "cache"),
 	}
-	err := oc.EnsureIsCached()
-	if err != nil {
+	if err := oc.EnsureIsCached(); err != nil {
 		glog.Errorln("Error starting the cluster: ", err)
 		atexit.Exit(1)
 	}
 
+	// Update instance-data.json for oc path
+	instancedata.Data.OcPath = filepath.Join(oc.GetCacheFilepath(), constants.OC_BINARY_NAME)
+	if err := instancedata.Data.WriteInstanceData(); err != nil {
+		glog.Warningf("Error updating oc path in instance-data.json file: ", err)
+	}
```

Comment:
Would you not have to return the error in this case and abort the provisioning? You are just logging the error. If we cannot write this, things will fail further down the road where we rely on getting the OcPath